### PR TITLE
Loading and storing raw data buffers in the database

### PR DESCRIPTION
**Pull request by scottmmorris**
### Description

This PR will change the data buffers to be loaded and stored into the db using the raw parameter to avoid the unnecessary cost of JSON serialisation and deserialisation

### Tasks

1. [x] Introduce the raw parameter into the data transaction calls to the db
2. [x] Run the new benchmark tests with the update

### CODE REVIEW

File: src/inodes/INodeManager.ts [hunk 1]
```diff
@@ -743,14 +743,14 @@ class INodeManager {
     for await (const data of dataDb.createReadStream(options)) {
       // This is to account for the case where a some blocks are missing in a database
       // i.e. blocks 0 -> 3 have data and a write operation was performed on blocks 7 -> 8
-      while (inodesUtils.unbufferId((data as any).key) !== blockCount) {
+      while (inodesUtils.unbufferId((data as any).key) != blockCount) {
```
Comment (CMCDragonkai) [inline src/inodes/INodeManager.ts:746]: Why did you change this? The `!==` should be used for anything that is not a null comparison.

File: src/inodes/INodeManager.ts [hunk 2]
```diff
@@ -743,14 +743,14 @@ class INodeManager {
...
         yield Buffer.alloc(blockSize);
       }
-      const plainTextData = await this.db.deserializeDecrypt<string>(
+      const plainTextData = await this.db.deserializeDecrypt(
```
Comment (CMCDragonkai) [inline src/inodes/INodeManager.ts:749]: Does `raw` being `true` meant that we don't use the generic type parameter? I think it's because we're always expecting a `Buffer` right?

**Comment by CMCDragonkai** [comment]
All tests passed and lintfixed?

File: src/inodes/INodeManager.ts (hunk 1 above)
Comment (scottmmorris) [reply src/inodes/INodeManager.ts:746]: I copied some of the code over so that must have been in the old code, Ill fix that now

File: src/inodes/INodeManager.ts (hunk 2 above)
Comment (scottmmorris) [reply src/inodes/INodeManager.ts:749]: Yeah if it is raw then it will always return a buffer

File: src/inodes/INodeManager.ts [hunk 3]
```diff
@@ -769,15 +769,15 @@ class INodeManager {
...
     for await (const data of dataDb.createReadStream(options)) {
       key = inodesUtils.unbufferId((data as any).key);
-      value = await this.db.deserializeDecrypt<string>(
+      value = await this.db.deserializeDecrypt(
         (data as any).value as Buffer,
-        false,
+        true,
       );
     }
-    if (value == null || key == null) {
```
Comment (scottmmorris) [inline src/inodes/INodeManager.ts:777]: Im fixing this up too, changing it back to null

**Comment by scottmmorris** [comment]
Yep, linted and all tests passing with the latest commit

**Comment by CMCDragonkai** [comment]
Ok please squash as well. I'll add my bench marks on top.

**Comment by scottmmorris** [comment]
Have squashed

**Comment by CMCDragonkai** [comment]
New benchmarks!

image

image

image

**Comment by CMCDragonkai** [comment]
Compared to the master:

image

image

image

**Comment by CMCDragonkai** [comment]
Looks like it's about up to 10% improvement in get and put speeds.

**Comment by CMCDragonkai** [comment]
Sorry I just realised that the benchmarks shouldn't change much because we don't actually have EFS benchmarks yet. That's still pending due to #50. So ignore the above screencaps.
